test/hack/resource/pkg/resourcetypes/instanceprofile.go

```diff
@@ -44,39 +44,40 @@ func (ip *InstanceProfile) Global() bool {
 }
 
 func (ip *InstanceProfile) GetExpired(ctx context.Context, expirationTime time.Time, excludedClusters []string) (names []string, err error) {
-	out, err := ip.iamClient.ListInstanceProfiles(ctx, &iam.ListInstanceProfilesInput{})
+	instanceProfiles, err := ip.getAllInstanceProfiles(ctx)
 	if err != nil {
 		return names, err
 	}
 
-	errs := make([]error, len(out.InstanceProfiles))
-	for i := range out.InstanceProfiles {
+	errs := make([]error, len(instanceProfiles))
+	for i := range instanceProfiles {
 		profiles, err := ip.iamClient.ListInstanceProfileTags(ctx, &iam.ListInstanceProfileTagsInput{
-			InstanceProfileName: out.InstanceProfiles[i].InstanceProfileName,
+			InstanceProfileName: instanceProfiles[i].InstanceProfileName,
 		})
 		if err != nil {
 			errs[i] = err
 			continue
 		}
 
-		clusterName, _ := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
+		clusterName, foundClusterName := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
 			return lo.FromPtr(tag.Key) == karpenterTestingTag
 		})
 		// Checking to make sure we are only list resources in the given region
-		region, _ := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
-			return lo.FromPtr(tag.Key) == v1.LabelTopologyZone
+		region, foundRegion := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
+			return lo.FromPtr(tag.Key) == v1.LabelTopologyRegion
 		})
 
-		if slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value)) || lo.FromPtr(region.Value) != lo.Must(config.LoadDefaultConfig(ctx)).Region {
+		if (foundClusterName && slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value))) || (foundRegion && lo.FromPtr(region.Value) != lo.Must(config.LoadDefaultConfig(ctx)).Region) {
 			continue
 		}
 
 		for _, t := range profiles.Tags {
 			// Since we can only get the date of the instance profile (not the exact time the instance profile was created)
 			// we add a day to the time that it was created to account for the worst-case of the instance profile being created
 			// at 23:59:59 and being marked with a time of 00:00:00 due to only capturing the date and not the time
-			if lo.FromPtr(t.Key) == karpenterTestingTag && out.InstanceProfiles[i].CreateDate.Add(time.Hour*24).Before(expirationTime) {
-				names = append(names, lo.FromPtr(out.InstanceProfiles[i].InstanceProfileName))
+			if lo.FromPtr(t.Key) == karpenterTestingTag && instanceProfiles[i].CreateDate.Add(time.Hour*24).Before(expirationTime) {
```

Comment:
Seems like we're doing a bit of double-work here. We first already check whether the cluster name is in our set of excluded clusters which means we should have been able to validate that this is a "karpenter-owned" resource above, so we really don't need to check this again here. We could really avoid all of the iteration through this tags section and just reduce all of this to a few "found" checks.